app/src/main/java/org/oppia/app/help/HelpViewModel.kt

```diff
@@ -1,9 +1,32 @@
 package org.oppia.app.help
 
-import androidx.lifecycle.ViewModel
-import org.oppia.app.fragment.FragmentScope
+import androidx.appcompat.app.AppCompatActivity
+import org.oppia.app.viewmodel.ObservableViewModel
+import java.text.FieldPosition
 import javax.inject.Inject
 
-/** [ViewModel] for [HelpFragment]. */
-@FragmentScope
-class HelpViewModel @Inject constructor() : ViewModel()
+/** [ObservableViewModel] for the recycler view of HelpActivity. */
+class HelpViewModel @Inject constructor(
+  private var activity: AppCompatActivity
+) : ObservableViewModel(), HelpNavigator {
+  public var title = ""
+  public var position: Int = -1
+
+  constructor(category: String, position: Int, activity: AppCompatActivity) : this(activity) {
```

Comment:
Remove this constructor and instead directly pass title inside the helpViewModel by removing the dagger injection. Something like this:

`class HelpViewModel(val title: String) : ObservableViewModel()`

Reply:
@rt4914 

`class HelpViewModel (val title: String) : ObservableViewModel() {

}`

I tried doing this, but always this error is coming :

`error: [Dagger/MissingBinding] org.oppia.app.help.HelpViewModel cannot be provided without an @Inject constructor or an @Provides-annotated method.`

Instead if I write,

`class HelpViewModel @Inject constructor(val title: String) : ObservableViewModel() {

}`

Then this error is coming,

`error: [Dagger/MissingBinding] java.lang.String cannot be provided without an @Inject constructor or an @Provides-annotated method`

Any way to bypass this error? Or Shall I go with the secondary constructor?

Reply:
The error that you are getting is because you are using that ViewModel in `viewModelProvider` in presenter. As mentioned earlier in one of the comments, remove that because you are not using it.

something like this:
```
package org.oppia.app.help

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import org.oppia.app.viewmodel.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import org.oppia.app.R
import org.oppia.app.databinding.HelpFragmentBinding
import org.oppia.app.fragment.FragmentScope
import javax.inject.Inject

/** The presenter for [HelpFragment]. */
@FragmentScope
class HelpFragmentPresenter @Inject constructor(
  private val activity: AppCompatActivity,
  private val fragment: Fragment
) {
  private var recyclerView: RecyclerView? = null
  private var helpCategoryAdapter: HelpCategoryAdapter? = null
  var arrayList = ArrayList<HelpViewModel>()
  private lateinit var binding: HelpFragmentBinding

  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
    binding = HelpFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
    // NB: Both the view model and lifecycle owner must be set in order to correctly bind LiveData elements to
    // data-bound view models.
    binding.let {
      it.lifecycleOwner = fragment
    }
    recyclerView = binding.root.findViewById(R.id.help_fragment_recycler_view) as? RecyclerView
    helpCategoryAdapter = HelpCategoryAdapter(activity, getRecyclerViewItemList())
    recyclerView!!.setLayoutManager(LinearLayoutManager(activity))
    recyclerView!!.setAdapter(helpCategoryAdapter)
    return binding.root
  }

  private fun getRecyclerViewItemList(): ArrayList<HelpViewModel> {
    for (dir in HelpItems.values()) {
      if (dir.equals(HelpItems.FAQ)) {
        val category1 = fragment.getString(R.string.frequently_asked_questions_FAQ)
        val helpViewModel: HelpViewModel = HelpViewModel(category1)
        arrayList.add(helpViewModel)
      }
    }
    return arrayList
  }
}

```

Reply:
Okay thanks a lot.